test/cache.bats

```diff
@@ -57,14 +56,15 @@ setup() {
   export RUBY_BUILD_SKIP_MIRROR=
   local checksum="ba988b1bb4250dee0b9dd3d4d722f9c64b2bacfc805d1b6eba7426bda72dd3c5"
 
-  stub shasum true "echo invalid" "echo $checksum"
+  stub shasum true true "echo invalid" true "echo $checksum"
```

Comment:
Part of the refactor was removing the `HAS_X_SUPPORT` variables (where X is md5, sha2, etc). Previously, each potential checksum algorithm was tested for regardless whether it would ever need to be used. Now there's a general `has_checksum_support` function. This refactor simultaneously reduced in some areas but also increased in others the number of times an algorithm was tested for existence. In cases where the checksum algo is never needed, it's never checked for support. (e.g. https://github.com/rbenv/ruby-build/pull/881/files#diff-9fa50c58f7de44021ff445803deb773cL13) However, since there is no longer a variable the saves off the result of the support test, algo support is tested repeatedly in some areas.

To resolve this, I can update the `has_checksum_support` function to memoize the results which should remove the necessity for all those extra stubbings.